ppx/ppx.ml

```diff
@@ -576,15 +576,32 @@ let jsxMapper () =
           let propsWithName =
             List.filter (fun (name, _) -> getLabel name != "") nonEmptyProps
           in
+          let unionLocation first second =
+            let diff = Location.compare first second in
+            let loc_start = {first.loc_start with pos_lnum= diff} in
+            let loc_end = second.loc_end in
+            {loc_start; loc_end; loc_ghost= false}
+          in
           let makePropField (arg_label, value) =
+            let valueLoc = value.pexp_loc in
+            let argLabelLoc = unionLocation loc value.pexp_loc in
```

Comment:
I append both locations to ensure the argLabel is part of the error.

Before
```
File "example/src/Events.re", line 49, characters 6-13:
49 |       <button type_="submit" xxxx=0> {"submit dis" |> React.string} </button>
           ^^^^^^^
Error: prop 'xxxx' isn't a valid React prop
```

Passing the expression loc
```
49 |       <button type_="submit" xxxx=0> {"submit dis" |> React.string} </button>
                                       ^
Error: prop 'xxxx' isn't a valid React prop
```

After (with unioning the locations)
```
File "example/src/Events.re", lines 1-49, characters 6-35:
-1 |       <button type_="submit" xxxx=0> {"submit dis" |> React.string} </button>
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: prop 'xxxx' isn't a valid React prop
```

Reply:
Hope it's clear. We could get fancier: with the length of the argLabel and minus the expr loc minus 1 (the `=`) maybe can point to the argLabel. Let me know if you think is worth it.

Reply:
This is great! Another idea (for the future ™️ ) could be to print after the error a link to a dedicated page in https://ml-in-barcelona.github.io/jsoo-react/ that explains native elements, and lists the whole list of tags that are allowed (the list could be shared code with the ppx ✨ ).

One nice thing is that because all these checks happen at compile time we don't need to care about the errors or urls making bundle size larger.

Reply:
> I append both locations to ensure the argLabel is part of the error.

Would it be easier to pick the location from the parent ast node?

Reply:
I mean, the `Pexp_apply` one:

<img width="880" alt="image" src="https://user-images.githubusercontent.com/220424/143505822-2245a757-f1a4-4f0f-b338-8877a0763c7a.png">

Reply:
That's way smarter!

Pushed a commit with this change, and appended a small refactor afterwards.